### PR TITLE
Enabling Local Development on Macs with Apple Silicon (arm64) 🍏

### LINKED ISSUE #1273: Enabling Local Development on Macs with Apple Silicon (arm64) 🍏💻
Navigating the intricacies of running our solution on Macs with Apple Silicon (M1 and M2) has been a journey. Unfortunately, SQL Express doesn't support Macs, particularly those with Apple Silicon chips. Fear not, fellow developers! I'm creating this issue and a PR to ensure that all you wonderful folks with Macs M1 or M2 can seamlessly run the application locally 🚀

### PR DESCRIPTION
**Pull request by david-acm**
Fixes #1273 

- Adds a docker-compose file with two services and a network. One service for the SQL database, and one for the web project.
- Since `docker compose` starts the web application together with the database, it becomes necessary to run migrations on the newly created database as part of the program startup code. I created an extensions class that runs migrations only if both:
  - The running environment is Local
  - And there are no pending migrations to be run in the local database

### CODE REVIEW

File: src/DevBetterWeb.Web/DevBetterWeb.Web.csproj [hunk 1]
```diff
@@ -19,6 +19,9 @@
 		<Content Update="wwwroot\**\*;Views\**\*;Areas\**\Views;appsettings.json;web.config">
 			<CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
 		</Content>
+		<Content Include="..\..\.dockerignore">
+		  <Link>.dockerignore</Link>
+		</Content>
```
Comment (KyleMcMaster) [inline src/DevBetterWeb.Web/DevBetterWeb.Web.csproj:24]: Is this section required?

Comment (david-acm) [reply src/DevBetterWeb.Web/DevBetterWeb.Web.csproj]: It's not. I removed it in the last commit just now

File: src/DevBetterWeb.Web/appsettings.Local.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
+{
+	"googleReCaptcha:SiteKey": "6LcEwN4ZAAAAAC-_UfJ0dmdU8bRxLLJ4FwjXyE1R",
+	"googleReCaptcha:SecretKey": "YOUR_SECRET_KEY",
+	"storageconnectionstring": "[connection string goes here]",
+	"videoUrlPrefix": "https://devbetter2storage.blob.core.windows.net/videos/",
+	"GoogleMapsAPIKey": "GOOGLE_MAPS_API_KEY",
+	"ConnectionStrings": {
+		"DefaultConnection": "Server=database,1433;MultipleActiveResultSets=true;User Id=sa;Password=bigStr0ngPwd*;Encrypt=false"
```
Comment (KyleMcMaster) [inline src/DevBetterWeb.Web/appsettings.Local.json:8]: Are these secrets safe to store in source control? Why store appsettings.Local.json in source control at all? Wouldn't it be better to ignore this file and then you wouldn't have to worry about accidently checking in secrets when you update it locally.

File: src/DevBetterWeb.Web/Program.cs [hunk 3]
```diff
@@ -195,14 +196,16 @@
...
 
 app.Run();
 
-
 static async Task SeedDatabase(IHost host)
 {
 	using var scope = host.Services.CreateScope();
+
+
 	var services = scope.ServiceProvider;
```
Comment (KyleMcMaster) [inline src/DevBetterWeb.Web/Program.cs:209]:
```suggestion
	using var scope = host.Services.CreateScope();
	var services = scope.ServiceProvider;
```

File: src/DevBetterWeb.Web/Program.cs [hunk 4]
```diff
@@ -212,6 +215,7 @@ static async Task SeedDatabase(IHost host)
 	var userManager = services.GetRequiredService<UserManager<ApplicationUser>>();
 	SeedData.PopulateInitData(context, userManager);
 
+	
 	if (environment == "Production")
```
Comment (KyleMcMaster) [inline src/DevBetterWeb.Web/Program.cs:219]:
```suggestion
	SeedData.PopulateInitData(context, userManager);

	if (environment == "Production")
```

**Comment by KamRon-67** [comment]
Was able to get everything up and running no problem. 

image

**Comment by david-acm** [comment]
Let me add some instructions to the README before this goes to main.

**Comment by david-acm** [comment]
@KyleMcMaster I removed the 'Local' appsettings file and environment [in this commit](https://github.com/DevBetterCom/DevBetterWeb/pull/1274/commits/25809fc6cd06d0db1de9d8279a9d93d8e5c2b0de) in favor of a docker environment variable. So, now the app behaves differently only if the 'DOTNET_RUNNING_IN_CONTAINER' env variable is set to true, which is set by either the IDE or the dotnet runtime (not sure which), no need to set it manually.

If that variable is set to true and the environment to 'Development' the app runs the DB migrations. You can also check this proposed workflow in the [Run with docker section in the README.md](https://github.com/DevBetterCom/DevBetterWeb/pull/1274/commits/25809fc6cd06d0db1de9d8279a9d93d8e5c2b0de)

**Comment by david-acm** [comment]
@KyleMcMaster @ardalis. This PR has been open for a while, I believe it has no pending issues. Should we run workflows and merge it?

**Comment by ardalis** [comment]
@KyleMcMaster when you get a chance can you confirm this is good to 🚢 ?

**Comment by KyleMcMaster** [comment]
> @KyleMcMaster when you get a chance can you confirm this is good to 🚢 ?

Looks fine to me, I was just casually reviewing. I say :shipit:!

**Comment by david-acm** [comment]
I think this PR has been missed for a while. I just updated the branch from main. I'm looking forward to contribute to some of the open issues but can't run the DB on my Mac without this. @ardalis could you merge it?

**Comment by ardalis** [comment]
@david-acm done!

**Pull request merged**